storage/database.go

```diff
@@ -120,9 +141,24 @@ func NewDatabase(shardSet m3db.ShardSet, opts m3db.DatabaseOptions) Database {
 		openCh:       make(chan struct{}, 1),
 		doneCh:       make(chan struct{}, dbOngoingTasks),
 	}
-	database.bsm = newBootstrapManager(database)
-	database.fm = newFlushManager(database)
-	return database
+	d.bsm = newBootstrapManager(d)
+	d.fm = newFlushManager(d)
+
+	var err error
+	if d.commitLog, err = commitlog.NewCommitLog(opts); err != nil {
+		return nil, err
+	}
+
+	switch opts.GetCommitLogStrategy() {
```

Comment:
might be cleaner to add a `Validate()` method to the options interface and call `opts.Validate()` instead, which not only checks this but also things like bufferPast should be smaller than block size etc.